### PR TITLE
Merge Signing Component into Master

### PR DESCRIPTION
**Pull request by GuyInFridge**
- Add 'embed photo' button to formatting buttons
- Sync scrolling of 'Input' to 'preview' textboxes (one way, input -> preview only)

### CODE REVIEW

File: src/app/pages/signing/signing.component.css [hunk 1]
```diff
@@ -131,6 +136,31 @@ markdown {
...
+#embed-photo-dialog, .modal-content {
+  max-width: 69em;
+}
+
+#embed-modal {
+  z-index: 2000;
+}
+
+#width-slider {
+  height: 10px;
```
Comment (janakitti) [inline src/app/pages/signing/signing.component.css]: Must this be in `px`?

File: src/app/pages/signing/signing.component.html [hunk 2]
```diff
@@ -169,3 +178,99 @@ <h5>Emojis</h5>
...
+        <div class="col-md-6 my-3 px-5 vertical-center">
+          <div class="mb-3">
+            <span class="mr-3"><strong>Add a link to a photo!</strong></span>
+            <button class="btn-sm btn-primary" data-toggle="modal" data-target="#embed-modal">
+              Need help?
+            </button>
+          </div>
+          <div class="input-group">
+            <label for="embed-link-field" style="display:none">Insert embed link</label>
+            <input class="form-control" type="text" placeholder="Put your link here!"
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.html:254]: `placeholder` gets cut off on mobile

File: src/app/pages/signing/signing.component.html [hunk 3]
```diff
@@ -169,3 +178,99 @@ <h5>Emojis</h5>
     </div>
   </div>
 </div>
+
+<!-- Picture not showing up? Modal -->
+<div class="modal fade" id="embed-modal" tabindex="-1" role="dialog" aria-labelledby="embed-modal-title"
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.html:188]: - `embed-modal` does not look good on mobile
- `embed-modal` should not be in dark theme

File: src/app/pages/signing/signing.component.css (hunk 1 above)
Comment (DabeerAzeez) [inline src/app/pages/signing/signing.component.css]:
```suggestion
  height: 0.625em;
```

File: src/app/pages/signing/signing.component.ts [hunk 4]
```diff
@@ -1,10 +1,11 @@
-import { Component, HostListener, OnInit } from '@angular/core';
+import { Component, ElementRef, HostListener, OnInit } from '@angular/core';
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.ts]: Unused import
```suggestion
import { Component, HostListener, OnInit } from '@angular/core';
```

File: src/app/pages/signing/signing.component.ts [hunk 5]
```diff
@@ -19,6 +20,11 @@ import { Title } from '@angular/platform-browser';
 export class SigningComponent implements OnInit {
   public name = 'your friend';
   public sending = false;
+  public embedLink = '';
+  public imageWidth = 50;
+  public copyButtonText = 'Copy';
+  public scrollSyncLoc = 0;
+  public signScrolling = true;
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.ts]: Unused symbols
```suggestion
```

File: src/app/pages/signing/signing.component.ts [hunk 6]
```diff
@@ -51,6 +57,12 @@ export class SigningComponent implements OnInit {
     return !this.signingService.signingBoxText || this.sending;
   }
 
+  syncMarkdown(textbox: HTMLTextAreaElement, previewBox: MarkdownComponent) {
+    const scrollPerc: number = (textbox.scrollTop + 160) / textbox.scrollHeight;
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.ts]: Don't know what "Perc" is, but it's marked as a typo

File: src/app/pages/signing/signing.component.html [hunk 7]
```diff
@@ -169,3 +178,105 @@ <h5>Emojis</h5>
...
+              <input #EmbedCode type="text" class="form-control"
+                     value="<img src='{{embedLink}}' width={{imageWidth}}%>">
+              <div class="input-group-append">
+                <button class="btn btn-primary" type="button" (click)="copyLink(EmbedCode)">{{copyButtonText}}</button>
+              </div>
+            </div>
+          </div>
+        </div>
+        <div class="col-md-6 mb-3 mb-md-0" id="embed-image-preview align-items-center">
+          <p><strong>Image Preview</strong></p>
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.html]: The image preview should be centred in the box, and it should fill the entire box. Ideally, the box would resize to the aspect ratio of the image with a limitation on the max dimensions.

Comment (DabeerAzeez) [reply src/app/pages/signing/signing.component.html]: That's not what I was going for, and that wouldn't be as helpful. Left aligning with the box being a constant width shows the user what the image will actually look like within their message (I wouldn't want them to have it be centered and pretty looking in the preview only for them to paste it into their message and have it look different.

File: src/app/pages/signing/signing.component.ts [hunk 8]
```diff
@@ -51,6 +61,12 @@ export class SigningComponent implements OnInit {
     return !this.signingService.signingBoxText || this.sending;
   }
 
+  syncMarkdown(textbox: HTMLTextAreaElement, previewBox: MarkdownComponent) {
```
Comment (DabeerAzeez) [inline src/app/pages/signing/signing.component.ts:69]: @TommyHuang-dev this function requires some documentation, please :)

File: src/app/pages/signing/signing.component.ts (hunk 6 above)
Comment (DabeerAzeez) [reply src/app/pages/signing/signing.component.ts]: I think `Perc` and `ele` could be renamed to be more clear, @TommyHuang-dev

**Comment by DabeerAzeez** [comment]
@TommyHuang-dev please redo your latest commit but pull this time.

File: src/app/pages/signing/signing.component.css [hunk 9]
```diff
@@ -135,6 +140,46 @@ markdown {
...
+#embed-image-card {
+  border: 0.0625em solid grey;
+  min-height: calc(1.5em + .75rem + 2px);
+  color: grey;
+}
+
+#embed-link-field {
+  border: 0.1em solid var(--accent);
+  border-radius: 0.2em;
+}
```
Comment (janakitti) [inline src/app/pages/signing/signing.component.css:162]:
```suggestion
}

```

File: src/app/pages/signing/signing.component.css [hunk 10]
```diff
@@ -135,6 +140,46 @@ markdown {
...
+#embed-photo-div {
+  text-align: center;
+}
+
+#embed-photo-dialog, .modal-content {
+  max-width: 69em;
```
Comment (janakitti) [inline src/app/pages/signing/signing.component.css:169]: nice

File: src/app/pages/signing/signing.component.html [hunk 11]
```diff
@@ -176,18 +184,123 @@ <h5>Emojis</h5>
...
+        <ol>
+          <li>Upload your photo to some service like Google Drive or Imgur. Or just find a photo that's already
+            online.
+          </li>
+          <li>Open the photo in your browser and see if you can right click the photo and click 'Open Image in New
+            Tab'
+          </li>
+          <li>Keep doing this until the website that you are on contains ONLY the image you want to embed, with no other
+            buttons or logos
+            or anything around it. If you see this sort of page, then copy the URL as that URL will now work as an
```
Comment (janakitti) [inline src/app/pages/signing/signing.component.html]:
```suggestion
            or anything around it. Once you see this sort of page, then copy the URL as that URL will now work as an
```

**Pull request merged**